src/app/cases/cases-routing.module.ts

```diff
@@ -25,32 +25,32 @@ const routes: Routes = [
         path: 'case/:caseId',
         component: CaseComponent,
         children: [
-          { path: 'details', component: CaseDetailsComponent, canActivate: [LeaveGuard] },
-          { path: 'person', component: CasePersonComponent, canActivate: [LeaveGuard] },
+          { path: 'details', component: CaseDetailsComponent, canDeactivate: [LeaveGuard] },
+          { path: 'person', component: CasePersonComponent, canDeactivate: [LeaveGuard] },
           {
             path: 'hospitalization',
             component: CaseHospitalizationComponent,
-            canActivate: [LeaveGuard],
+            canDeactivate: [LeaveGuard],
           },
           {
             path: 'port-health',
             component: CasePortHealthComponent,
-            canActivate: [LeaveGuard],
+            canDeactivate: [LeaveGuard],
           },
-          { path: 'symptoms', component: CaseSymptomsComponent, canActivate: [LeaveGuard] },
+          { path: 'symptoms', component: CaseSymptomsComponent, canDeactivate: [LeaveGuard] },
           {
             path: 'epidemiological-data',
             component: CaseEpidemiologicalDataComponent,
-            canActivate: [LeaveGuard],
+            canDeactivate: [LeaveGuard],
           },
-          { path: 'therapy', component: CaseTherapyComponent, canActivate: [LeaveGuard] },
-          { path: 'follow-up', component: CaseFollowUpComponent, canActivate: [LeaveGuard] },
+          { path: 'therapy', component: CaseTherapyComponent, canDeactivate: [LeaveGuard] },
+          { path: 'follow-up', component: CaseFollowUpComponent, canDeactivate: [LeaveGuard] },
           {
             path: 'clinical-course',
             component: CaseClinicalCourseComponent,
-            canActivate: [LeaveGuard],
+            canDeactivate: [LeaveGuard],
```

Comment:
is this change correct?

Reply:
This is the logical issue that was found by @danielrus , and I mentioned in the standup. This was changed everywhere we used LeaveGuard. Places where it was Guard are unchanged and still use canActivate like before.